monai/apps/reconstruction/dictionary.py

```diff
@@ -0,0 +1,301 @@
+            image = d[key]
+            roi_size = d["target"].shape[1:]
+            roi_center = tuple(i // 2 for i in image.shape[1:])
+            cropper = SpatialCrop(roi_center=roi_center, roi_size=roi_size)
+            d[key] = convert_to_tensor(cropper(d[key]))
+        return d
+
+
+class InputTargetNormalizeIntensityd(MapTransform):
+    """
+    Dictionary-based wrapper of
+    :py:class:`monai.transforms.NormalizeIntensity`.
+    This is similar to :py:class:`monai.transforms.NormalizeIntensityd`
+    but is different since (1) it automatically normalizes the target based
+    on the input data, and (2) it also returns mean and std after normalization. The
+    first dimension is reserved for channel (e.g., could be num_slices).
+    The rest of the dimensions should be spatial. So the shape is (C,H,W,D)
+    for 3D samples.
+
+    Args:
+        keys: keys of the corresponding items to be transformed.
+            See also: monai.transforms.MapTransform
+        subtrahend: the amount to subtract by (usually the mean)
+        divisor: the amount to divide by (usually the standard deviation)
+        nonzero: whether only normalize non-zero values.
+        channel_wise: if True, calculate on each channel separately,
+            otherwise, calculate on the entire image directly. default
+            to False.
+        dtype: output data type, if None, same as input image. defaults
+            to float32.
+        allow_missing_keys: don't raise exception if key is missing.
+
+    Note:
+        "target" (which holds the ground-truth image) is a required key, no need
+        to pass it as one of the "keys". Once the input is normalized, "target"
+        will be normalized according to the mean-std of the input.
+    """
+
+    backend = NormalizeIntensity.backend
+
+    def __init__(
+        self,
+        keys: KeysCollection,
+        subtrahend: Optional[NdarrayOrTensor] = None,
+        divisor: Optional[NdarrayOrTensor] = None,
+        nonzero: bool = False,
+        channel_wise: bool = False,
+        dtype: DtypeLike = np.float32,
+        allow_missing_keys: bool = False,
+    ) -> None:
+        super().__init__(keys, allow_missing_keys)
+        self.default_normalizer = NormalizeIntensity(subtrahend, divisor, nonzero, channel_wise, dtype)
+
+    def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, NdarrayOrTensor]:
+        """
+        this will normalize the input first, and then according to its mean-std,
+        it will normalize the target.
+
+        Args:
+            data: is a dictionary containing (key,value) pairs from
+                the loaded dataset
+
+        Returns:
+            the new data dictionary
+
+        Note:
+            "target" is a required key in the data which holds the groun-truth image..
+        """
+        d = dict(data)
+        for key in self.key_iterator(d):
+            # compute mean of each slice in the input for mean-std normalization
```

Comment:
So this transform is slice-wise NormalizeIntensity?

Reply:
What is the difference between slice and channel? I'm confused

Reply:
Not really, it has a dual functionality. Let me make some adjustments to clarify this.

Reply:
Just clarified this and added channel-wise functionality explicitly to the code.